sphinx/builders/linkcheck.py

```diff
@@ -460,7 +460,7 @@ def check_uri() -> Tuple[str, str, int]:
                                                  config=self.config, auth=auth_info,
                                                  **kwargs)
                         response.raise_for_status()
-                    except (HTTPError, TooManyRedirects) as err:
+                    except (HTTPError, TooManyRedirects, ConnectionError) as err:
```

Comment:
IMO, a comment would help explaining `ConnectionError`. If the connection couldn’t be made to the server, the question “why retry as GET” is legitimate and a comment would help answering it.